### PR TITLE
Add examples for Azure Disk Encryption Extension for VM and VMSS

### PR DESCRIPTION
**Pull request by myc2h6o**
VM Examples are from https://github.com/hashicorp/terraform-provider-azurerm/pull/15356/files
Adding examples for VMSS as well

### CODE REVIEW

File: quickstart/201-vm-disk-encryption-extension/provider.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
+terraform {
```
Comment (grayzu) [inline quickstart/201-vm-disk-encryption-extension/provider.tf]: rename to providers.tf

File: quickstart/201-vm-disk-encryption-extension/variables.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+variable "vm_public_key" {
+  type        = string
+  description = "Public key of the Virtual Machine"
+}
+
+variable "name_prefix" {
+  type        = string
+  default     = "tftest"
```
Comment (grayzu) [inline quickstart/201-vm-disk-encryption-extension/variables.tf]: Remove default value.

File: quickstart/201-vmss-disk-encryption-extension/provider.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
+terraform {
```
Comment (grayzu) [inline quickstart/201-vmss-disk-encryption-extension/provider.tf]: rename to providers.tf

File: quickstart/201-vmss-disk-encryption-extension/variables.tf [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+variable "location" {
+  type        = string
+  description = "Location where resources will be created"
+}
+
+variable "name_prefix" {
+  type        = string
+  default     = "tftest"
```
Comment (grayzu) [inline quickstart/201-vmss-disk-encryption-extension/variables.tf]: remove default value.

**Review by grayzu — CHANGES_REQUESTED** [review]
looks great! few minor comments.

File: quickstart/201-vm-disk-encryption-extension/main.tf [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    "Get",
+    "Delete",
+    "Set",
+  ]
+}
+
+resource "azurerm_key_vault_key" "example" {
+  name         = "examplekey"
+  key_vault_id = azurerm_key_vault.example.id
+  key_type     = "RSA"
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/main.tf]: An idea inspired by [BridgeCrew Checkov](https://www.checkov.io/)'s [policy `CKV_AZURE_112`](https://docs.bridgecrew.io/docs/ensure-that-key-vault-key-is-backed-by-hsm):

>For added assurance, when you use Azure Key Vault, you can import or generate keys in hardware security modules (HSMs) that never leave the HSM boundary. This scenario is often referred to as bring your own key, or BYOK. Azure Key Vault uses nCipher nShield family of HSMs (FIPS 140-2 Level 2 validated) to protect your keys.

Would the `key_type` assign to `RSA-HSM` be better?

File: quickstart/201-vm-disk-encryption-extension/main.tf [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+// Disk Encryption Extension
+resource "azurerm_virtual_machine_extension" "example" {
+  name                       = "AzureDiskEncryptionForLinux"
+  publisher                  = "Microsoft.Azure.Security"
+  type                       = "AzureDiskEncryptionForLinux"
+  type_handler_version       = "1.1"
+  auto_upgrade_minor_version = false
+  virtual_machine_id         = azurerm_linux_virtual_machine.example.id
+
+  settings = <<SETTINGS
```
Comment (lonegunmanb) [inline quickstart/201-vm-disk-encryption-extension/main.tf]: The `settings` looks like a json string, would using `jsonencode` function be better here?

File: quickstart/201-vmss-disk-encryption-extension/main.tf [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    "Get",
+    "Delete",
+    "Set",
+  ]
+}
+
+resource "azurerm_key_vault_key" "example" {
+  name         = "examplekey"
+  key_vault_id = azurerm_key_vault.example.id
+  key_type     = "RSA"
```
Comment (lonegunmanb) [inline quickstart/201-vmss-disk-encryption-extension/main.tf]: Should we consider using `RSA-HSM` here?

File: quickstart/201-vmss-disk-encryption-extension/main.tf [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+// Disk Encryption Extension
+resource "azurerm_virtual_machine_scale_set_extension" "example" {
+  name                         = "AzureDiskEncryption"
+  publisher                    = "Microsoft.Azure.Security"
+  type                         = "AzureDiskEncryption"
+  type_handler_version         = "2.2"
+  auto_upgrade_minor_version   = false
+  virtual_machine_scale_set_id = azurerm_windows_virtual_machine_scale_set.example.id
+
+  settings = <<SETTINGS
```
Comment (lonegunmanb) [inline quickstart/201-vmss-disk-encryption-extension/main.tf]: Maybe we can use `jsonencode` here?

**Comment by myc2h6o** [comment]
Hi @grayzu and @lonegunmanb thanks for reviewing the pr! I've resolved the comments from both of you, could you take another look? Update in the change:
- renamed provider.tf to provider**s**.tf
- removed the default value for `name_prefix`
- use `RSA-HSM` for the key vault key
- use `jsonencode` to specify the json settings

**Comment by grayzu** [comment]
LGTM

**Pull request merged**